assembly/src/ast/mod.rs

```diff
@@ -177,11 +193,17 @@ impl ProgramAst {
         // asserts below are OK because we enforce limits on the number of procedure and the
         // number of body instructions in relevant parsers
 
-        assert!(self.local_procs.len() <= u16::MAX as usize, "too many local procs");
+        assert!(self.imports.len() <= MAX_LOCAL_PROCS, "too many imports");
+        target.write_u16(self.imports.len() as u16);
+        // We don't need to serialized the keys, since they are the last name in the path
+        let values: Vec<LibraryPath> = self.imports.clone().into_values().collect();
+        values.write_into(&mut target);
```

Comment:
nit: to avoid extra cloning, I'd probably just have a loop here which iterates over values and writes them into the target one-by-one (should be just one extra line of code).